### PR TITLE
Add cargo clippy to CI

### PR DESCRIPTION
**Pull request by nhynes**

### CODE REVIEW

File: src/oid.rs [hunk 1]
```diff
@@ -16,32 +16,24 @@ impl Oid {
         Oid(s.to_owned())
     }
 
-    /// Convert the OID to a string representation.
-    /// The string contains the IDs separated by dots, for ex: "1.2.840.113549.1.1.5"
-    pub fn to_string(&self) -> String {
```
Comment (chifflier) [inline src/oid.rs]: Why was this function removed? Is it related to clippy?

Comment (nhynes) [reply src/oid.rs]: Yes. `to_string` is a provided method of `Display`, so it hasn’t been removed—just moved.

Comment (chifflier) [reply src/oid.rs]: Right, thanks for the reply

**Comment by chifflier** [comment]
Merged in 4e5e0b06a3412face56b3cdbbfe1e2e7691dd301 with modifications in 7680d8eef7ddbb7a668e2668b152280eefa38262
I also added a check for rustfmt
Thanks!

**Pull request closed without merging**